### ISSUE TITLE
Accessibility: Non-semantic Elements: Edit permalink button in main editor area should be a button (not div)

### ISSUE THREAD

**Issue by alisterscott** [Automattic/wp-calypso#1995]
This issue has been split out from: https://github.com/Automattic/wp-calypso/issues/697

Edit permalink button in main editor area should be a `button` because keyboard users cannot access `div` elements. They aren't keyboard accessible by default.

**Comment by hafizrahman** [comment]
I researched this and found two alternatives:

1) Keep them as `div` and add `role="button"` and `tabindex="0"` to it. This makes the element accessible to screen readers. However, it still doesn't help with keyboard users, as they will be able to focus to it but hitting enter/space will not trigger the `div`. We will need to add specific keyboard handler to read keypresses and hook it up to `onClick`.

or,

2) Replace `div` with `button`. This keeps the feature working in my test, and theoretically it will help both screen reader and keyboard users. I'm just not yet sure if it's a good HTML practice to have other elements inside the `button` element like in our case.

This is my source: https://www.marcozehe.de/2013/04/24/easy-aria-tip-6-making-clickables-accessible/

Button elements can have content according to this article: https://css-tricks.com/use-button-element/, and so far I think the second option is the way to go. I just think I need to confirm this further before making a PR.

**Comment by lancewillett** [comment]
Needs attention before we can move ahead with it — ping @hafizrahman for a comment on the pull request related: #3800.

**Comment by designsimply** [comment]
#bug-scrub #pre-gutenburg-bug-purge

Tested and confirmed that the edit permalink icon is still in a div and cannot be tabbed to and activated with the "enter key.

<img width="1280" alt="screen shot 2018-01-05 at fri jan 5 4 54 02 pm" src="https://user-images.githubusercontent.com/1119271/34633489-a660771c-f239-11e7-8d1e-b6a578a127c3.png">
<sup>Seen at https://wordpress.com/post/madefortesting.com/1 using Firefox 57.0.3 on macOS 10.13.2.</sup>
